Composer/packages/client/src/pages/skills/skill-settings.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import React, { useState, useEffect } from 'react';
+import formatMessage from 'format-message';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+
+import { DialogSetting, BoundAction } from '../../store/types';
+
+import { FormFieldAlignHorizontalBotSettings } from './styles';
+
+export interface ISkillFormProps {
+  botId?: string;
+  skillHostEndpoint?: string;
+  setSettings: BoundAction;
+  projectId: string;
+  botName: string;
+  settings: DialogSetting;
+}
+
+const SkillSettings: React.FC<ISkillFormProps> = props => {
+  const [skillSettings, setSkillSettings] = useState({
+    botId: props.botId,
+    skillHostEndpoint: props.skillHostEndpoint,
+  });
+  useEffect(() => {
+    setSkillSettings({
+      botId: props.botId,
+      skillHostEndpoint: props.skillHostEndpoint,
+    });
+  }, [props.botId, props.skillHostEndpoint]);
+
+  const handleFieldChange = event => {
+    const localSettings = {
+      ...skillSettings,
+      [event.target.id]: event.target.value,
+    };
+    setSkillSettings({ ...localSettings });
+    props.setSettings(props.projectId, props.botName, { ...props.settings, ...localSettings }, undefined);
+  };
```

Comment:
Question: is it necessary to keep the skill settings in state? Does `props.setSettings` not update them through the props?

It's not clear to me why have the `useState` as well as a `useEffect` with the same props.

Reply:
I had this here for navigating in/out of the skills page or a hard refresh on the skills page. maybe it isn't necessary